pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -677,13 +677,21 @@ class CreateVmModal extends React.Component {
             const stateDelta = { [key]: value };
             if (value && value.recommendedResources.ram) {
                 stateDelta.recommendedMemory = value.recommendedResources.ram;
-                this.onValueChanged('memorySize', convertToUnit(stateDelta.recommendedMemory, units.B, this.state.memorySizeUnit));
+                const converted = Math.floor(convertToUnit(stateDelta.recommendedMemory, units.B, this.state.memorySizeUnit));
+                if (converted == 0)
+                    this.setState({ memorySizeUnit: units.MiB.name, memorySize: Math.floor(convertToUnit(stateDelta.recommendedMemory, units.B, units.MiB)) });
+                else
+                    this.onValueChanged('memorySize', converted);
```

Comment:
to make sure I understand this correctly - when you convert value and it is less than 1<this.state.memorySizeUnit> then it must be MiBs and it will be at least 1MiB, right?
This works because there are only GiB and MiB and the memory must be always at least 1MiB?

Reply:
So, that recommended memory value comes from osinfo-db and I have never seen any to be less than 1G to be honest. It's really the recommended value to value for an operating system so < 1G is a bit crazy anyway.
However, in the test VMs we have limited resources, and in order that I will skip the dialog error about insufficient memory when spawning new VMs, I so tweak little bit the values in osinfo db, making the recommended values 256M. So 1MiB recommended memory for OS installation is not a things ever :P

Reply:
CirrOS, Alpine, or similar "micro-Linuxes" should be fine with 128 MiB even -- that should certainly not be a recommendation, but it's a rather plausible minimum.

Reply:
I am not sure what you suggest here @martinpitt. 
There is indeed differentiation between recommended values and minimum allowed values.
The memory entry is autofilled with the recommended values and not the minumum in case these are available, else 1G is the default values.

JFYI just checked and libosinfo does not currently have recommended memory value for CirroOS.
see
[kkoukiou@sequoia cockpit-bak]$ pkg/machines/getOSList.py | jq | grep -B3 -A13 'CirrOS 0.3.4'
  {
    "id": "http://cirros-cloud.net/cirros/0.3.4",
    "shortId": "cirros0.3.4",
    "name": "CirrOS 0.3.4",
    "version": "0.3.4",
    "family": "linux",
    "vendor": "CirrOS",
    "releaseDate": "2015-04-22",
    "eolDate": "2017-02-14",
    "codename": "",
    "recommendedResources": {},
    "minimumResources": {
      "ram": 134217728,
      "storage": 52428800
    },
    "treeInstallable": false
  },

Reply:
@KKoukiou : I just meant that there are plausible cases where recommended or minimal values are < 1 GiB. I. e. this commit is a nice fix. :-)